README.md

```diff
@@ -12,194 +12,129 @@ shared. These resources have a different lifecycle than resources in workspaces.
 
 ## Implementing a Landing Zone
 
-### Landing Zone Definition Factories and Landing Zone Definitions.
+Landing zone creation process is implemented as a [Stairway](https://github.com/DataBiosphere/stairway) flight represented by the following class `CreateLandingZoneFlight`. This flight consists of different steps one of which is step which 
+represents sub-flight to create Azure resources. This sub-flight is represented by the following class `CreateLandingZoneResourcesFlight`. This class utilizes specific steps' provider to define the list of steps required to create Azure resources.
 
-Landing zones are implemented using the factory pattern; the factory creates *Landing Zone Definitions* (LZDs).
+Below are lists of changes required to introduce new landing zone:
+1) Implement specific steps which are responsible for Azure resource creation (see example below for `CreateVnetStep`)
+2) Implement specific steps' definition provider (by implementing `StepsDefinitionProvider` interface)
+3) Introduce new landing zone type by extending `StepsDefinitionFactoryType`
+4) Update mapping at `LandingZoneStepsDefinitionProviderFactory` based on landing zone type from section 3. This mapping is used by sub-flight to get flights' steps to create specific Azure resources.
 
-Landing Zone Definitions are where resources and their purpose are defined.
+In case of updating existing landing zone it is required to do following:
+1) Introduce new or adjusting existing step
+2) In case of a new step it is also required to include it into corresponding steps' provider
 
-A Landing Zone Definition factory is an implementation of:
+### Landing Zone Flight Step Providers
 
+Landing zones are implemented using providers classes which return list of necessary steps to create a landing zone. Each provider should implement following interface `StepsDefinitionProvider`
 ```java
-public interface LandingZoneDefinitionFactory {
-    DefinitionHeader header();
+public interface StepsDefinitionProvider {
```

Comment:
Instead of duplicating the definition here, I would suggest just linking to the file. I think there needs to be documentation explaining `landingZoneDefintion`, `header`, and `availableVersions` are, and that should be JavaDoc within the source file itself.